fmriprep/config.py

```diff
@@ -627,3 +627,39 @@ def init_spaces(checkpoint=True):
 
     # Make the SpatialReferences object available
     workflow.spaces = spaces
+
+
+def load_previous_config(output_dir, work_dir, participant=None):
+    """
+    Search for existing config file, and carry over previous options.
+    If a participant label is specified, the output directory is searched for
+    the most recent config file. If no config is found, the working directory is searched.
+
+    Returns
+    -------
+    success : bool
+        Existing config found and loaded
+
+    """
+    conf = None
+    if participant:
+        # output directory
+        logdir = Path(output_dir) / 'fmriprep' / f'sub-{participant[0]}' / 'log'
+        for f in logdir.glob(f'**/{CONFIG_FILENAME}'):
+            conf = conf or f
+            if f.stat().st_mtime > conf.stat().st_mtime:
+                conf = f
+
+    if conf is None:
+        # work directory
+        conf = Path(work_dir) / f'.{CONFIG_FILENAME}'
+        if not conf.exists():
+            conf = None
+
+    # load existing config
+    if conf is not None:
+        print(f"Reusing config file {conf}")
+        load(conf)
```

Comment:
this should load in the old uuid. Why don't you just override it here? (instead of within the parser)

Reply:
we could, but then we'd have the logic in two separate places (here + attribute definition).

in the parser, it is hit once regardless of the presence of an existing config